lib/fog/ovirt/requests/compute/v4/storage_domains.rb

```diff
@@ -0,0 +1,25 @@
+module Fog
+  module Compute
+    class Ovirt
+      class V4
+        class Real
+          def storage_domains(filter = {})
+            client.system_service.storage_domains_service.list.collect do |sd|
```

Comment:
Except for roles, there are other search options, remove roles from filter, and pass the filter to `list`. After that you can deal with roles in code.